python/lsst/sims/featureScheduler/schedulers/core_scheduler.py

```diff
@@ -0,0 +1,195 @@
+
+        self.log = logging.getLogger("Core_scheduler")
+        # initialize a queue of observations to request
+        self.queue = []
+        # Are the observations in the queue part of a sequence.
+        self.queue_is_sequence = False
+        # The indices of self.survey_lists that provided the last addition(s) to the queue
+        self.survey_index = [None, None]
+
+        # If we have a list of survey objects, convert to list-of-lists
+        if isinstance(surveys[0], list):
+            self.survey_lists = surveys
+        else:
+            self.survey_lists = [surveys]
+        self.nside = nside
+        hpid = np.arange(hp.nside2npix(nside))
+        self.ra_grid_rad, self.dec_grid_rad = _hpid2RaDec(nside, hpid)
+        # Should just make camera a class that takes a pointing and returns healpix indices
+        if camera == 'LSST':
+            self.pointing2hpindx = hp_in_lsst_fov(nside=nside)
+        elif camera == 'comcam':
+            self.pointing2hpindx = hp_in_comcam_fov(nside=nside)
+        else:
+            raise ValueError('camera %s not implamented' % camera)
+
+        # keep track of how many observations get flushed from the queue
+        self.flushed = 0
+
+    def flush_queue(self):
+        """"
+        Like it sounds, clear any currently queued desired observations.
+        """
+        self.queue = []
+        self.queue_is_sequence = False
+        self.survey_index = [None, None]
+
+    def add_observation(self, observation):
+        """
+        Record a completed observation and update features accordingly.
+
+        Parameters
+        ----------
+        observation : dict-like
+            An object that contains the relevant information about a
+            completed observation (e.g., mjd, ra, dec, filter, rotation angle, etc)
+        """
+
+        # Find the healpixel centers that are included in an observation
+        indx = self.pointing2hpindx(observation['RA'], observation['dec'], observation['rotSkyPos'])
+        for surveys in self.survey_lists:
+            for survey in surveys:
+                survey.add_observation(observation, indx=indx)
+
+    def update_conditions(self, conditions_in):
+        """
+        Parameters
+        ----------
+        conditions : dict-like
+            The current conditions of the telescope (pointing position, loaded filters, cloud-mask, etc)
+        """
+        # Add the current queue and scheduled queue to the conditions
+        self.conditions = conditions_in
+        # put the local queue in the conditions
+        self.conditions.queue = self.queue
+
+        # XXX---TODO:  Could potentially put more complicated info from all
+        # the surveys in the conditions object here. e.g., when a DDF plans to next request
+        # observations.
+
+    def check_queue_mjd_only(self, mjd):
+        """
+        Check if there are things in the queue that can be executed using only MJD and not full conditions
+        """
```

Comment:
Yes, this is a dirty little secret method that probably should only be called by `sim_runner` (let's it know it can safely skip generating a fresh `Conditions` object). Leading underscore added.